File: shared/types/calendar.ts
/** 投资日历（calendar_event）类型：camelCase wire（CamelModel）+ camelCase 领域类型。 */

export const CALENDAR_CATEGORIES = ['宏观', '央行动态', '新股', '解禁', '财报', '会议'] as const

export type CalendarEventCategory = (typeof CALENDAR_CATEGORIES)[number]

/** 后端 GET /calendar/events 响应项（camelCase wire）。 */
export interface ApiCalendarEventResponse {
  id: number
  eventTime: string
  endTime: string | null
  title: string
  category: string
  impactMarkets: string[] | null
  source: string | null
  sourceUrl: string | null
  relatedSymbols: string[] | null
}

/** 日历事件领域类型（camelCase，前端使用）。 */
export interface CalendarEvent {
  id: number
  eventTime: string
  endTime: string | null
  title: string
  category: CalendarEventCategory
  impactMarkets: string[]
  source: string | null
  sourceUrl: string | null
  relatedSymbols: string[]
}

File: backend/app/schemas/calendar.py
"""投资日历的 Pydantic schemas。"""

from datetime import datetime

from app.schemas.base import CamelModel


class CalendarEventResponse(CamelModel):
    """投资日历事件的响应 schema。"""

    id: int
    event_time: datetime
    end_time: datetime | None = None
    title: str
    category: str
    impact_markets: list[str] | None = None
    source: str | None = None
    source_url: str | None = None
    related_symbols: list[str] | None = None
